common/src/main/cpp/classes/openxr_fb_scene_manager.cpp

```diff
@@ -0,0 +1,390 @@
+#include "classes/openxr_fb_spatial_entity_query.h"
+
+using namespace godot;
+
+void OpenXRFbSceneManager::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_default_scene", "packed_scene"), &OpenXRFbSceneManager::set_default_scene);
+	ClassDB::bind_method(D_METHOD("get_default_scene"), &OpenXRFbSceneManager::get_default_scene);
+
+	ClassDB::bind_method(D_METHOD("set_scene_setup_method", "method_name"), &OpenXRFbSceneManager::set_scene_setup_method);
+	ClassDB::bind_method(D_METHOD("get_scene_setup_method"), &OpenXRFbSceneManager::get_scene_setup_method);
+
+	ClassDB::bind_method(D_METHOD("set_auto_create", "enable"), &OpenXRFbSceneManager::set_auto_create);
+	ClassDB::bind_method(D_METHOD("get_auto_create"), &OpenXRFbSceneManager::get_auto_create);
+
+	ClassDB::bind_method(D_METHOD("set_visible", "visible"), &OpenXRFbSceneManager::set_visible);
+	ClassDB::bind_method(D_METHOD("get_visible"), &OpenXRFbSceneManager::get_visible);
+	ClassDB::bind_method(D_METHOD("show"), &OpenXRFbSceneManager::show);
+	ClassDB::bind_method(D_METHOD("hide"), &OpenXRFbSceneManager::hide);
+
+	ClassDB::bind_method(D_METHOD("create_scene_anchors"), &OpenXRFbSceneManager::create_scene_anchors);
+	ClassDB::bind_method(D_METHOD("remove_scene_anchors"), &OpenXRFbSceneManager::remove_scene_anchors);
+	ClassDB::bind_method(D_METHOD("are_scene_anchors_created"), &OpenXRFbSceneManager::are_scene_anchors_created);
+
+	ClassDB::bind_method(D_METHOD("request_scene_capture", "request"), &OpenXRFbSceneManager::request_scene_capture, DEFVAL(""));
+	ClassDB::bind_method(D_METHOD("is_scene_capture_enabled"), &OpenXRFbSceneManager::is_scene_capture_enabled);
+
+	ClassDB::bind_method(D_METHOD("get_anchor_uuids"), &OpenXRFbSceneManager::get_anchor_uuids);
+	ClassDB::bind_method(D_METHOD("get_anchor_node", "uuid"), &OpenXRFbSceneManager::get_anchor_node);
+	ClassDB::bind_method(D_METHOD("get_spatial_entity", "uuid"), &OpenXRFbSceneManager::get_spatial_entity);
+
+	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "default_scene", PROPERTY_HINT_RESOURCE_TYPE, "PackedScene"), "set_default_scene", "get_default_scene");
+	ADD_PROPERTY(PropertyInfo(Variant::STRING_NAME, "scene_setup_method", PROPERTY_HINT_NONE, ""), "set_scene_setup_method", "get_scene_setup_method");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "auto_create", PROPERTY_HINT_NONE, ""), "set_auto_create", "get_auto_create");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "visible", PROPERTY_HINT_NONE, ""), "set_visible", "get_visible");
+
+	ADD_SIGNAL(MethodInfo("scene_anchor_created", PropertyInfo(Variant::Type::OBJECT, "scene_node"), PropertyInfo(Variant::Type::OBJECT, "spatial_entity")));
+	ADD_SIGNAL(MethodInfo("scene_data_missing"));
+	ADD_SIGNAL(MethodInfo("scene_capture_completed", PropertyInfo(Variant::Type::BOOL, "success")));
+}
+
+bool OpenXRFbSceneManager::_set(const StringName &p_name, const Variant &p_value) {
+	PackedStringArray parts = p_name.split("/", true, 2);
+	if (parts[0] == "scenes") {
```

Comment:
Do we need to check that `parts.size() == 2`?

Reply:
Ah, yes, we do, thanks! Updated in my latest push